### PR TITLE
Dialog: automatically add clicked callback for the standard button

### PR DESCRIPTION
**Pull request by ogoffart**
And make the viewer listen to them and set them to automatically close dialogs

### CODE REVIEW

File: docs/builtin_elements.md [hunk 1]
```diff
@@ -657,7 +657,13 @@ A Dialog should have one main element for the content, that is not a button.
...
 The button will be layed out in an order that depends on the platform.
 
-The `kind` property of the `StandardButton`s needs to be set to a specific value. It cannot be a complex expression.
+The `kind` property of the `StandardButton`s needs to be set to a specific value. It cannot be a complex expression,
+and there cannot be several button of the same kind.
+
+If A callback `<kind>_clicked` is automatically added for each button which does not have an explicit
+callback handler, so it can be handled from the native code. (e.g. if there is a button of kind `cancel`,
+a `cancel_clicked` callback will be added)
+When viewed with the `sixtyfps-viewer` program, the `ok`, `cancel`, and `close` button will cause the dialog to close.
```
Comment (tronical) [inline docs/builtin_elements.md:666]: Soon we'll need a README.md for the viewer :)

File: tools/viewer/main.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ struct Cli {
     #[structopt(name = "path to .60 file", parse(from_os_str))]
     path: std::path::PathBuf,
 
-    /// The style name ('native', or 'ugly')
+    /// The style name ('native', or 'ulgy')
```
Comment (tronical) [inline tools/viewer/main.rs:31]:
```suggestion
    /// The style name ('native', or 'ugly')
```

This was fixed in 21f9850b4378cd05e80455619701524d59575b72 :)

**Review by tronical — APPROVED** [review]

File: tools/viewer/main.rs (hunk 2 above)
Comment (ogoffart) [reply tools/viewer/main.rs:31]: i don't know why this was overwritten, maybe a did a pull and i had some unsaved changed in that file. vscode did not show me a warning

File: docs/builtin_elements.md (hunk 1 above)
Comment (ogoffart) [reply docs/builtin_elements.md:666]: yes

Comment (ogoffart) [reply docs/builtin_elements.md:666]: Done: https://github.com/sixtyfpsui/sixtyfps/blob/master/tools/viewer/README.md

**Pull request merged**